person_main.rb

```diff
@@ -0,0 +1,47 @@
+require_relative './teacher'
+require_relative './student'
+
+class Personmain
+  def list_all_people(people)
+    people.each { |n| puts n.to_string }
+  end
+
+  def create_a_person(people)
+    puts 'Do you want to create a Student (1) or a Teacher (2)? [Input the number]: '
+    var = gets.chomp
+    case var
+    when '1'
+      create_a_student(people)
+    when '2'
+      create_a_teacher(people)
+    end
+    puts 'Person created Succesfully'
+  end
+
+  def create_a_student(people)
+    default_classroom = Classroom.new('default')
+    puts 'Age: '
+    age = gets.chomp
+    puts 'Name: '
+    name = gets.chomp
+    puts 'Has parent permission? [Y/N]: '
+    permission = gets.chomp
+    student = if permission.upcase == 'Y'
+                Student.new(age, default_classroom, name, parent_permission: true)
+              else
+                Student.new(age, default_classroom, name, parent_permission: false)
+              end
+    people.push(student)
+  end
+
+  def create_a_teacher(people)
+    puts 'Age: '
+    age = gets.chomp
+    puts 'Name: '
+    name = gets.chomp
+    puts 'Specilaization'
+    specialization = gets.chomp
```

Comment:
- [ ] Your requests for both `age` and `name` in the **`create_a_student`** and **`create_a_teacher`** methods here break the DRY rule. Kindly extract these requests into one method and call the method in both method instead.

- [ ] The request for `permission` and its validation should be handled by another method

- [ ] The request for `specialization` should be handled by another method.